ReactWindows/ReactNative.Shared/UIManager/BorderedCanvas.cs

```diff
@@ -47,6 +52,298 @@ public Border Border
+            {
+                var borderProps = GetBorderProps();
+                var borderBrush = (borderProps?.Color.HasValue ?? false) ? new SolidColorBrush(borderProps.Color.Value) : DefaultBorderBrush;
+                Border = new Border { BorderBrush = borderBrush };
+
+                // Layout animations bypass SetDimensions, hence using XAML bindings.
+
+                Border.SetBinding(FrameworkElement.WidthProperty, new Binding
+                {
+                    Source = this,
+                    Path = new PropertyPath("Width")
+                });
+
+                Border.SetBinding(FrameworkElement.HeightProperty, new Binding
+                {
+                    Source = this,
+                    Path = new PropertyPath("Height")
+                });
+            }
+
+            return Border;
+        }
+
+        /// <summary>
+        /// Gets or sets background brush of Canvas itself or of Border child as
+        /// appropriate. This is the only supported method to access background
+        /// of BorderedCanvas. DO NOT use Canvas.Background or Border.Background directly.
+        /// </summary>
+        public new Brush Background
+        {
+            get
+            {
+                if (Border == null)
+                {
+                    return base.Background;
+                }
+                else
+                {
+                    var border = GetOrCreateBorder();
+                    return border.Background;
+                }
+            }
+            set
+            {
+                if (Border == null)
+                {
+                    base.Background = value;
+                }
+                else
+                {
+                    var border = GetOrCreateBorder();
+                    border.Background = value;
+                    base.Background = base.Background ?? _defaultBackgroundBrush;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the border radius of the view.
+        /// </summary>
+        public CornerRadius BorderRadius
+        {
+            get
+            {
+                if (HasBorder())
+                {
+                    var border = GetOrCreateBorder();
+                    return border.CornerRadius;
+                }
+                else
+                {
+                    return new CornerRadius();
+                }
+            }
+            set
+            {
+                var border = GetOrCreateBorder();
+                border.CornerRadius = value;
+                TransferBackgroundBrush();
+            }
+        }
+
+        /// <summary>
+        /// Sets the background color of the view.
+        /// </summary>
+        public Color BackgroundColor
+        {
+            set => Background = new SolidColorBrush(value);
+        }
+
+        /// <summary>
+        /// Set the border color of the view.
+        /// </summary>
+        public Color? BorderColor
```

Comment:
>BorderColor [](start = 22, length = 11)

Why BorderColor and not BorderBrush?

Reply:
Current API only allows to set color of border. Goal of this PR is just to adapt code for linear gradient change. Maybe for new features it's better to have separate PR?

---
In reply to: [178548707](https://github.com/Microsoft/react-native-windows/pull/1730#discussion_r178548707) [](ancestors = 178548707)

Reply:
I think of the API surfaced by <View/> as different from this BorderedCanvas helper, which I think should be more closely aligned with XAML - should we just expose the BorderBrush on this?